### PR TITLE
Bluetooth: Mesh: correction lpn rx delay

### PR DESCRIPTION
**Pull request by alxelax**
If extended advertiser is used, then advertiser sends callback
quite close to the real tx end time. No necessity to consider
advertisement time in lpn rx delay. It allows lpn to spend
less time in active scanning and save more power. Unfortunately,
legacy advertiser shows very slow precision. It is mandatory to
consider adv time for it. Additionally, unnecessary considering of
the advertisement duration has been removed for receive window in lpn.
It makes scanner window longer for a couple of milliseconds.
This is the responsibility of friend node to configure window that
it can manage.

Signed-off-by: Aleksandr Khromykh <aleksandr.khromykh@nordicsemi.no>

### CODE REVIEW

**Review by LingaoM — DISMISSED** [review]

File: subsys/bluetooth/mesh/lpn.c [hunk 1]
```diff
@@ -38,6 +38,12 @@ LOG_MODULE_REGISTER(bt_mesh_lpn);
 	 CONFIG_LOG_MODE_DEFERRED Kconfig option when Low Power node feature is enabled.
 #endif
 
+#if BT_MESH_ADV_LEGACY
```
Comment (PavelVPV) [inline subsys/bluetooth/mesh/lpn.c:41]:
```suggestion
#if CONFIG_BT_MESH_ADV_LEGACY
```

**Review by PavelVPV — COMMENTED** [review]
I don't like adding dependency on type of advertiser in lpn code, but I can't give better solution.

**Review by Andrewpini — DISMISSED** [review]

File: subsys/bluetooth/mesh/lpn.c [hunk 2]
```diff
@@ -911,8 +916,7 @@ static void lpn_timeout(struct k_work *work)
 		clear_friendship(true, false);
 		break;
 	case BT_MESH_LPN_RECV_DELAY:
-		k_work_reschedule(&lpn->timer,
-				  K_MSEC(lpn->adv_duration + SCAN_LATENCY + lpn->recv_win));
+		k_work_reschedule(&lpn->timer, K_MSEC(SCAN_LATENCY + lpn->recv_win));
```
Comment (PavelVPV) [inline subsys/bluetooth/mesh/lpn.c:919]: If Receive Delay was started `RX_DELAY_CORRECTION` ms earlier, shouldn't Receive Window be extended by this time (like with `SCAN_LATENCY`)?

Comment (alxelax) [reply subsys/bluetooth/mesh/lpn.c:919]: sounds reasonable

**Review by PavelVPV — DISMISSED** [review]

**Review by PavelVPV — DISMISSED** [review]

File: subsys/bluetooth/mesh/lpn.c [hunk 3]
```diff
@@ -38,6 +38,12 @@ LOG_MODULE_REGISTER(bt_mesh_lpn);
 	 CONFIG_LOG_MODE_DEFERRED Kconfig option when Low Power node feature is enabled.
 #endif
 
+#if defined(CONFIG_BT_MESH_ADV_LEGACY)
+#define RX_DELAY_CORRECTION ((int32_t)lpn->adv_duration)
```
Comment (jhedberg) [inline subsys/bluetooth/mesh/lpn.c:42]: `lpn` should be a macro parameter here. Macros shouldn't make assumptions of the environment in which they're called (e.g. the availability or naming of specific variables in the current scope).

File: subsys/bluetooth/mesh/lpn.c [hunk 4]
```diff
@@ -912,7 +917,7 @@ static void lpn_timeout(struct k_work *work)
 		break;
 	case BT_MESH_LPN_RECV_DELAY:
 		k_work_reschedule(&lpn->timer,
-				  K_MSEC(lpn->adv_duration + SCAN_LATENCY + lpn->recv_win));
+			K_MSEC(SCAN_LATENCY + lpn->recv_win + RX_DELAY_CORRECTION));
```
Comment (jhedberg) [inline subsys/bluetooth/mesh/lpn.c:920]: minor: you're unnecessarily (and incorrectly) changing the indentation/alignment of the line

**Review by jhedberg — CHANGES_REQUESTED** [review]
Functionally the change looks good to me, however I do have a few design/style related comments.

**Review by PavelVPV — DISMISSED** [review]

**Review by jhedberg — DISMISSED** [review]

**Comment by alxelax** [comment]
For some reason, CI got stuck for some tests forever on the previous commit hash. Now this is Ok. Need reapproval for PR.

**Review by PavelVPV — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Review by Andrewpini — APPROVED** [review]

**Pull request merged**